src/components/HeaderComponent/HeaderThemeSwitcher/HeaderThemeSwitcher.tsx

```diff
@@ -0,0 +1,28 @@
+import { FC, useState, useContext } from "react";
+import { ThemeContext } from "../../../utils/providers/ThemeProvider";
+import { ThemeContextType } from "../../index"
+
+import "./HeaderThemeSwitcher.scss"
+
+import sun from "../../../assets/images/sun.svg"
+import moon from "../../../assets/images/moon.svg"
+
+
+export const HeaderThemeSwitcher: FC = () => {
+    // @ts-ignore
+    const [theme, setTheme] = useContext<ThemeContextType>(ThemeContext)
+    const [image, setImage] = useState(theme === "Light" ? moon : sun)
```

Comment:
You don't need to have extra state here as the change will lead to re render it's better to have a conditional rendering like `theme === 'dark' ? src to dark icon : src to light icon'`